### DISCUSSION TITLE
zh/nixos-with-flakes/nixos-with-flakes-enabled

### DISCUSSION THREAD

**Discussion by giscus** [ryan4yin/nixos-and-flakes-book#163] [Announcements]
# zh/nixos-with-flakes/nixos-with-flakes-enabled

一份非官方的新手指南

https://nixos-and-flakes.thiscute.world/zh/nixos-with-flakes/nixos-with-flakes-enabled

**Comment by Jiannan-dev** [comment]
抓个虫子，船新 => 全新

**Reply by NestorLiao** [reply]
这是个梗。船新=全新。来源于香港明星渣渣辉张家辉在为游戏《贪玩蓝月》做广告宣传时不标准的普通话发音。

**Comment by quanuanc** [comment]
这里有个疑问想问一下，flake.nix里的nixos分支
```
# NixOS 官方软件源，这里使用 nixos-23.11 分支
nixpkgs.url = "github:NixOS/nixpkgs/nixos-23.11";
```
如果后续出了更新版本的nixos，这个地方写的23.11是不是应该使用更新的？
例如，今天是2024-11-14，目前的nixos稳定版本号是24.05，那么这个时间安装的nixos默认版本是24.05，代码里的这个地方也应该写24.05吧？

**Reply by ryan4yin** [reply]
是的，你想用哪个版本，就改下这个版本号。

**Reply by AntonOfTheWoods** [reply]
我在用24.11，到现在没遇到问题！

**Comment by skymkmk** [comment]
能否理解为在当下，configuration.nix 主要配置和系统相关的选项，而 flake.nix 主要配置和软件包相关的内容？

**Reply by WangYiBen0** [reply]
不是吧，把 configuration.nix 里所有的东西写进 flake.nix 里也没有问题啊，只要启用了 flakes 特性

**Comment by yangjian226200** [comment]
hi.
我在运行这步
“nix flake init -t templates#full
cat flake.nix”
的时候没有自动在相应位置创建“/etc/nixos/flake.nix”
反而创建在这个位置“/home/用户/flake.nix”
然后豆包告诉我“cd /etc/nixos/”后“sudo nano flake.nix”可以创建。
我把代码复制进flake.nix并保存完成后运行“sudo nixos-rebuild switch”
才跟上进度，不知道搞错了没。

**Reply by zhalieheiye** [reply]
似乎这个命令是在当前目录下生成示范文件
你直接在/etc/nixos/里创建一个flake.nix文件，然后输入配置，也一样的
